### PR TITLE
Open shelf by default and shelf height settings

### PR DESCRIPTION
**Pull request by thomasslee97**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
This adds a setting to each rundown layout to set that the shelf should open by default when opening a rundown with that rundown layout. It also adds the ability to set a default shelf height, so if a height is not found in the local storage it will use that as the default, on a per layout basis. This setting can be left blank to fall back to the Sofie default. Finally, the shelf height is changed to be stored on a per-layout basis, as the comfortable viewing size for each layout may be different, so this allows users to switch between shelf layouts without the need to adjust the size every time they switch.

* **What is the current behavior?** (You can also link to an open issue here)
- Shelf is always hidden by default.
- Default shelf size is 50vh.
- Shelf size is stored "globally", meaning switching between layouts may require resizing the shelf to get a good fit for the shelf content.

* **What is the new behavior (if this is a feature change)?**
- Shelf can be set to open automatically when a particular rundown layout is in use (default is hidden).
- Default shelf size can be set on a per-layout basis (0 - 100, where 100 in the setting is full height (0vh)). This setting value is clamped to a max value (currently 95) as this prevents the user from being in a situation where they can't resize the shelf.
- Shelf size is stored on a per-layout basis using the layout `_id`.

* **Other information**:
Two new settings illustrated (Show Bucket setting is not part of this PR - screenshot taken from TV 2 core)
image

**Status**

- [ ] Code documentation for the relevant parts in the code have been added/updated by the PR author
- [X] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

**Comment by thomasslee97** [comment]
Created as draft until tested at TV 2 on their staging environment.

**Review by jstarpl — CHANGES_REQUESTED** [review]
Seems alright, but I have a question on the purpose of `shelfResizedByUser` in the RundownView state. It's not clear to me what that value is used for, since the interactions between `isInspectorShelfExpanded`, `shelfResizedByUsers` and `rundownLayout.openByDefault` seem to indicate to me that it's redundant.

File: meteor/client/ui/RundownView.tsx [hunk 1]
```diff
@@ -2604,7 +2613,10 @@ export const RundownView = translateWithTracker<IProps, IState, ITrackedProps>((
 								<ErrorBoundary>
 									<Shelf
 										buckets={this.props.buckets}
-										isExpanded={this.state.isInspectorShelfExpanded}
+										isExpanded={
+											this.state.isInspectorShelfExpanded ||
+											(!this.state.shelfResizedByUser && this.state.rundownLayout?.openByDefault)
```
[the remark this answers is not in the document]
Comment (thomasslee97) [reply meteor/client/ui/RundownView.tsx:2618]: If the selected layout changes or the layout data arrives late, it prevents the shelf from popping into view if the user has already chosen to hide it

[the remark this answers is not in the document]
Comment (jstarpl) [reply meteor/client/ui/RundownView.tsx:2608]: OK, now I get it. Could you perhaps add some code documentation on the State interface to clarify the purpose and perhaps change the name of the flag to `wasShelfResizedByUser` to indicate that this describes a past event and not a continuous action?

[the remark this answers is not in the document]
Comment (thomasslee97) [reply meteor/client/ui/RundownView.tsx]: I've made those changes

**Review by jstarpl — APPROVED** [review]

**Pull request merged**